### PR TITLE
Bugfixes

### PR DESCRIPTION
**Pull request by ayooluwaodemuyiwa**
Made LocationManagerSingleton so I don't have to write the same lines of code over and over again -- stops updating location when it is necessary to update. Fixed frame size so that it is relative to the view controller during my animation. Also, realized now that I should put an alert for when user doesn't have Apple Maps on their phone

### CODE REVIEW

File: HotSpot/View Controllers /CurrentAndPastDetails.m [hunk 1]
```diff
@@ -68,9 +72,31 @@ - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
...
+- (void) openMap{
+    Listing *listing = self.booking.listing;
+    [listing fetchInBackgroundWithBlock:^(PFObject * _Nullable object, NSError * _Nullable error){
+        CLLocationDegrees longitude = listing.address.longitude;
+        CLLocationDegrees latitude= listing.address.latitude;
+        NSString* directionsURL = [NSString stringWithFormat:@"http://maps.apple.com/?saddr=%f,%f&daddr=%f,%f", self.locationManager.locationManager.location.coordinate.latitude, self.locationManager.locationManager.location.coordinate.longitude,  latitude, longitude];
+        if ([[UIApplication sharedApplication] respondsToSelector:@selector(openURL:options:completionHandler:)]) {
+            [[UIApplication sharedApplication] openURL:[NSURL URLWithString: directionsURL] options:@{MKLaunchOptionsDirectionsModeKey:MKLaunchOptionsDirectionsModeDriving} completionHandler:^(BOOL success) {}];
+        } else {
+            NSLog(@"sorry, you need to have Apple Maps installed");
```
Comment (Akash30) [inline HotSpot/View Controllers /CurrentAndPastDetails.m:95]: remove the NSLogs. the user can't see it

File: HotSpot/View Controllers /LocationManagerSingleton.m [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        [self.locationManager setDistanceFilter:kCLDistanceFilterNone];
+        [self.locationManager setHeadingFilter:kCLHeadingFilterNone];
+        [self.locationManager requestWhenInUseAuthorization];
+        //do any more customization to your location manager
+    }
+    
+    return self;
+}
+
++ (LocationManagerSingleton*)sharedSingleton {
```
Comment (Akash30) [inline HotSpot/View Controllers /LocationManagerSingleton.m:39]: lol what's the difference between this method and "sharedLocationManager"? they do the same thing!

File: HotSpot/View Controllers /LocationManagerSingleton.m [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    if(!sharedSingleton) {
+        static dispatch_once_t onceToken;
+        dispatch_once(&onceToken, ^{
+            sharedSingleton = [LocationManagerSingleton new];
+        });
+                      }
+                      return sharedSingleton;
+                      }
+                      
+                      - (void)locationManager:(CLLocationManager *)manager didUpdateLocations:(nonnull NSArray<CLLocation *> *)locations {
```
Comment (Akash30) [inline HotSpot/View Controllers /LocationManagerSingleton.m:50]: oh gosh indentation please!

File: HotSpot/View Controllers/MainContainerViewController.h [hunk 4]
```diff
@@ -8,12 +8,15 @@
 
 #import <UIKit/UIKit.h>
 #import "MapKit/MapKit.h"
+#import <Speech/Speech.h>
 NS_ASSUME_NONNULL_BEGIN
 
-@interface MainContainerViewController : UIViewController <UISearchBarDelegate, MKLocalSearchCompleterDelegate>
+@interface MainContainerViewController : UIViewController <UISearchBarDelegate, MKLocalSearchCompleterDelegate, SFSpeechRecognizerDelegate, UISearchDisplayDelegate> {
+}
```
Comment (Akash30) [inline HotSpot/View Controllers/MainContainerViewController.h:15]: remove curly braces

File: HotSpot/View Controllers /LocationManagerSingleton.m (hunk 2 above)
Comment (ayooluwaodemuyiwa) [reply HotSpot/View Controllers /LocationManagerSingleton.m:39]: sorry didn't realize I wrote it twice

**Pull request closed without merging**